director/cache_ads.go

```diff
@@ -133,9 +133,18 @@ func recordAd(ctx context.Context, sAd server_structs.ServerAd, namespaceAds *[]
 
 	ad := server_structs.Advertisement{ServerAd: sAd, NamespaceAds: *namespaceAds}
 
-	customTTL := param.Director_AdvertisementTTL.GetDuration()
+	adTTL := time.Until(sAd.Expiration)
+	if sAd.Expiration.IsZero() {
+		adTTL = param.Director_AdvertisementTTL.GetDuration()
+		// Handle unit tests that do not initialize default config
+		if adTTL == 0 {
+			adTTL = 15 * time.Minute
```

Comment:
It'd be nice to log when this happens, just in case it manages to fire outside of a unit test.